File: ScoutTrack/ScoutTrack.UI/scouttrack_desktop/lib/models/city.dart
import 'package:json_annotation/json_annotation.dart';

part 'city.g.dart';

@JsonSerializable()
class City {
  final int id;
  final String name;
  final double latitude;
  final double longitude;
  final DateTime createdAt;
  final DateTime? updatedAt;
  final int troopCount;
  final int memberCount;

  City({
    this.id = 0,
    this.name = '',
    this.latitude = 0,
    this.longitude = 0,
    required this.createdAt,
    this.updatedAt,
    this.troopCount = 0,
    this.memberCount = 0,
  });

  factory City.fromJson(Map<String, dynamic> json) {
    return _$CityFromJson(json);
  }
}

File: ScoutTrack/ScoutTrack.Services/Database/Entities/City.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;
using System.Threading.Tasks;

namespace ScoutTrack.Services.Database.Entities
{
    public class City
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        [Required]
        public double Latitude { get; set; }

        [Required]
        public double Longitude { get; set; }

        public List<Troop> Troops { get; set; } = new List<Troop>();
        public List<Member> Members { get; set; } = new List<Member>();

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }
    }
}